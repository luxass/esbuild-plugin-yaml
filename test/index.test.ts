import { build } from 'esbuild';
import { describe, expect, it } from 'vitest';

import YAMLPlugin from '../src';

describe('Yaml Plugin', () => {
  it('build file', async () => {
    const result = await build({
      entryPoints: ['test/shared/config-utils.ts'],
      format: 'esm',
      bundle: true,
      plugins: [YAMLPlugin()],
      outfile: './test/shared/config-utils.js'
    });
    expect(result).toStrictEqual({ errors: [], warnings: [] });
  });
});
