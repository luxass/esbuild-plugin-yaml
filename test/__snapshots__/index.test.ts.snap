// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`handle multi documents yaml files 1`] = `
"var data = [
  {
    "apiVersion": "batch/v1",
    "kind": "CronJob",
    "metadata": {
      "name": "first-cronjob",
      "namespace": "default"
    },
    "spec": {
      "schedule": "*/10 * * * *",
      "concurrencyPolicy": "Forbid",
      "successfulJobsHistoryLimit": 3,
      "failedJobsHistoryLimit": 1,
      "jobTemplate": {
        "spec": {
          "template": {
            "spec": {
              "containers": [
                {
                  "name": "first-job",
                  "image": "busybox:latest",
                  "command": [
                    "/bin/sh",
                    "-c",
                    'echo "First job running"; date'
                  ]
                }
              ],
              "restartPolicy": "OnFailure"
            }
          }
        }
      }
    }
  },
  {
    "apiVersion": "batch/v1",
    "kind": "CronJob",
    "metadata": {
      "name": "second-cronjob",
      "namespace": "default"
    },
    "spec": {
      "schedule": "0 0 * * *",
      "concurrencyPolicy": "Forbid",
      "successfulJobsHistoryLimit": 3,
      "failedJobsHistoryLimit": 1,
      "jobTemplate": {
        "spec": {
          "template": {
            "spec": {
              "containers": [
                {
                  "name": "second-job",
                  "image": "busybox:latest",
                  "command": [
                    "/bin/sh",
                    "-c",
                    'echo "Second job running"; date'
                  ]
                }
              ],
              "restartPolicy": "OnFailure"
            }
          }
        }
      }
    }
  }
];
var cronjobs_default = data;

console.log(cronjobs_default);
"
`;
